custom_components/glinet/router.py

```diff
@@ -416,16 +425,17 @@ async def update_wireguard_client_state(self) -> None:
         response = await self._update_platform(self._api.wireguard_client_state)
         if not response:
             return
-        connected: bool = response["status"] == 1
-
-        # TODO in some circumstances this returns TypeError: 'NoneType' object is not subscriptable
-        if self._wireguard_clients[response["peer_id"]]:
-            client: WireGuardClient = self._wireguard_clients[response["peer_id"]]
-            client.connected = connected
-            if connected:
-                self._wireguard_connection = client
-                return
-        self._wireguard_connection = None
+        # 0 is disconnted, 1 is connected, 2 is connecting
+        self._wireguard_connections = []
+        for config in response:
+            connected: bool = config["status"] != 0
```

Comment:
Status does not exist when an item is disabled. This can be fixed with `config["enabled"] and config["status"] != 0`. I don't know if the enabled field exists in older versions though.